docs/Migration Guide.md

```diff
@@ -0,0 +1,152 @@
+# Migration Guide #
+
+This document describes how to port your code from using one ESR version
+of SpiderMonkey to the next ESR version.
+
+## ESR 60 to ESR 68 ##
+
+### Remove requests ###
+
+Previously, a "request" was a global lock on the JS interpreter that was
+required in order to use many parts of the JSAPI.
+This is no longer necessary, and can simply be deleted.
+
+**Recommendation**: Remove all `JSAutoRequest`, `JS_BeginRequest()`, and
+`JS_EndRequest()` from your code.
+
+### Rename compartments to realms ###
+
+In ESR 68, a "realm" is the term for the environment belonging to a
+particular global object, so when you enter the "realm" of a global
+object you are executing code with that object as the global object.
+As before, with compartments, objects from different realms cannot
+access each other unless they are "wrapped".
+
+JSAutoCompartment is renamed to JSAutoRealm, and CompartmentOptions are
+renamed to RealmOptions.
+Some APIs that previously operated explicitly on a global object, now
+operate on the current realm's global object, but there are currently no
+examples of that in this code.
+
+"Compartments" still exist, but they are now more focused on security,
+and are used for isolating code in different browser tabs in Firefox,
+for example.
+Realms are used within the same compartment for debugger code vs. client
+code, for example.
+All code in these examples is run within the same compartment.
```

Comment:
Does any of it run in different realms? If not, then this should say "...is run within the same realm."